modules/backendscheduler/cache.go

```diff
@@ -12,6 +12,9 @@ import (
 )
 
 func (s *BackendScheduler) flushWorkCache(ctx context.Context) error {
+	s.mtx.Lock()
```

Comment:
why add locking here? s.work seems to protect itself. as long as we're not worried about racy access to s.work i don't think we need this?